check_sites.py

```diff
@@ -70,8 +74,21 @@ def main():
         # If the file cannot be loaded, we will not run any other checks
             print(f"There was an error when loading {input_file};" 
                   f"\nerror was:  {inst}")
-            return  
-     
+            return
+        # Notify of any formatting errors in the JSON
+        if with_format:
+            f.seek(0)
+            loaded_file = f.read()
+            # Add final newline by convention
+            formatted_file = json.dumps(rws_sites, indent=2, ensure_ascii=False) + "\n"
+            if loaded_file != formatted_file:
+                diff = difflib.ndiff(loaded_file.splitlines(keepends=True), formatted_file.splitlines(keepends=True))
+                # Only show lines with differences
+                filtered_diff = filter(lambda line: len(line) > 0 and line[0] != ' ', diff)
```

Comment:
Generator comprehensions are "more pythonic" (https://google.github.io/styleguide/pyguide.html#210-lambda-functions).
```suggestion
                filtered_diff = (line for line in diff if len(line) > 0 and line[0] != ' ')
```